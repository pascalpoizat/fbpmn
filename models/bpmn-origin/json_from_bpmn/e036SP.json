{
    "isInterrupt": {},
    "catN": {
        "ExclusiveGateway_1449jmr": "AndGateway",
        "EndEvent_0a59w28": "NoneEndEvent",
        "EndEvent_1bu7x75": "NoneEndEvent",
        "StartEvent_1": "NoneStartEvent",
        "StartEvent_09ojxru": "NoneStartEvent",
        "P_": "Process",
        "EndEvent_0atuxkh": "NoneEndEvent",
        "SubProcess_1glz8ii": "SubProcess"
    },
    "catE": {
        "SequenceFlow_1ue74ws": "NormalSequenceFlow",
        "SequenceFlow_12utbfs": "NormalSequenceFlow",
        "SequenceFlow_00j4ong": "NormalSequenceFlow",
        "SequenceFlow_1yn4txk": "NormalSequenceFlow",
        "SequenceFlow_0xc489r": "NormalSequenceFlow"
    },
    "containN": {
        "P_": [
            "StartEvent_1",
            "SubProcess_1glz8ii",
            "EndEvent_0a59w28"
        ],
        "SubProcess_1glz8ii": [
            "StartEvent_09ojxru",
            "EndEvent_0atuxkh",
            "EndEvent_1bu7x75",
            "ExclusiveGateway_1449jmr"
        ]
    },
    "messageE": {},
    "timeInformation": {},
    "targetE": {
        "SequenceFlow_1ue74ws": "EndEvent_1bu7x75",
        "SequenceFlow_12utbfs": "SubProcess_1glz8ii",
        "SequenceFlow_00j4ong": "EndEvent_0a59w28",
        "SequenceFlow_1yn4txk": "ExclusiveGateway_1449jmr",
        "SequenceFlow_0xc489r": "EndEvent_0atuxkh"
    },
    "name": "e036SP",
    "attached": {},
    "containE": {
        "P_": [
            "SequenceFlow_12utbfs",
            "SequenceFlow_00j4ong"
        ],
        "SubProcess_1glz8ii": [
            "SequenceFlow_1yn4txk",
            "SequenceFlow_0xc489r",
            "SequenceFlow_1ue74ws"
        ]
    },
    "edges": [
        "SequenceFlow_12utbfs",
        "SequenceFlow_00j4ong",
        "SequenceFlow_1yn4txk",
        "SequenceFlow_0xc489r",
        "SequenceFlow_1ue74ws"
    ],
    "messages": [],
    "sourceE": {
        "SequenceFlow_1ue74ws": "ExclusiveGateway_1449jmr",
        "SequenceFlow_12utbfs": "StartEvent_1",
        "SequenceFlow_00j4ong": "SubProcess_1glz8ii",
        "SequenceFlow_1yn4txk": "StartEvent_09ojxru",
        "SequenceFlow_0xc489r": "ExclusiveGateway_1449jmr"
    },
    "nodes": [
        "P_",
        "StartEvent_1",
        "SubProcess_1glz8ii",
        "EndEvent_0a59w28",
        "StartEvent_09ojxru",
        "EndEvent_0atuxkh",
        "EndEvent_1bu7x75",
        "ExclusiveGateway_1449jmr"
    ],
    "nameN": {
        "SubProcess_1glz8ii": "SP"
    }
}