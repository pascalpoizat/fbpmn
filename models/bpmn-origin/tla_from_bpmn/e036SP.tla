---------------- MODULE e036SP ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "P_" :> {  }

ContainRel ==
  "P_" :> { "StartEvent_1", "SubProcess_1glz8ii", "EndEvent_0a59w28" }
  @@ "SubProcess_1glz8ii" :> { "StartEvent_09ojxru", "EndEvent_0atuxkh", "EndEvent_1bu7x75", "ExclusiveGateway_1449jmr" }

Node == { "P_", "StartEvent_1", "SubProcess_1glz8ii", "EndEvent_0a59w28", "StartEvent_09ojxru", "EndEvent_0atuxkh", "EndEvent_1bu7x75", "ExclusiveGateway_1449jmr" }

Edge == { "SequenceFlow_12utbfs", "SequenceFlow_00j4ong", "SequenceFlow_1yn4txk", "SequenceFlow_0xc489r", "SequenceFlow_1ue74ws" }

Message == {  }

msgtype ==
  [ i \in {} |-> {}]


source ==
   "SequenceFlow_12utbfs" :> "StartEvent_1"
@@ "SequenceFlow_00j4ong" :> "SubProcess_1glz8ii"
@@ "SequenceFlow_1yn4txk" :> "StartEvent_09ojxru"
@@ "SequenceFlow_0xc489r" :> "ExclusiveGateway_1449jmr"
@@ "SequenceFlow_1ue74ws" :> "ExclusiveGateway_1449jmr"

target ==
   "SequenceFlow_12utbfs" :> "SubProcess_1glz8ii"
@@ "SequenceFlow_00j4ong" :> "EndEvent_0a59w28"
@@ "SequenceFlow_1yn4txk" :> "ExclusiveGateway_1449jmr"
@@ "SequenceFlow_0xc489r" :> "EndEvent_0atuxkh"
@@ "SequenceFlow_1ue74ws" :> "EndEvent_1bu7x75"

CatN ==
   "P_" :> Process
@@ "StartEvent_1" :> NoneStartEvent
@@ "SubProcess_1glz8ii" :> SubProcess
@@ "EndEvent_0a59w28" :> NoneEndEvent
@@ "StartEvent_09ojxru" :> NoneStartEvent
@@ "EndEvent_0atuxkh" :> NoneEndEvent
@@ "EndEvent_1bu7x75" :> NoneEndEvent
@@ "ExclusiveGateway_1449jmr" :> Parallel

CatE ==
   "SequenceFlow_12utbfs" :> NormalSeqFlow
@@ "SequenceFlow_00j4ong" :> NormalSeqFlow
@@ "SequenceFlow_1yn4txk" :> NormalSeqFlow
@@ "SequenceFlow_0xc489r" :> NormalSeqFlow
@@ "SequenceFlow_1ue74ws" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

