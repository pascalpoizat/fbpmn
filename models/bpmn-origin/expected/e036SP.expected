Working in /tmp/e036SP.AYYlG with 1 worker(s)
[92mtransformation done[0m
<<"Processes=", 1, "Nodes=", 8, "Gateway=", 1, "SF=", 5, "MF=", 0>>
---------- Network01Bag ----------
[X] Prop01Type
     states=10 trans=12 depth=9
[X] Prop02Safe
     states=10 trans=12 depth=9
[X] Prop03Sound
     states=10 trans=12 depth=9
[X] Prop04MsgSound
     states=10 trans=12 depth=9
---------- Network02FifoPair ----------
[X] Prop01Type
     states=10 trans=12 depth=9
[X] Prop02Safe
     states=10 trans=12 depth=9
[X] Prop03Sound
     states=10 trans=12 depth=9
[X] Prop04MsgSound
     states=10 trans=12 depth=9
---------- Network03Causal ----------
[X] Prop01Type
     states=10 trans=12 depth=9
[X] Prop02Safe
     states=10 trans=12 depth=9
[X] Prop03Sound
     states=10 trans=12 depth=9
[X] Prop04MsgSound
     states=10 trans=12 depth=9
---------- Network04Inbox ----------
[X] Prop01Type
     states=10 trans=12 depth=9
[X] Prop02Safe
     states=10 trans=12 depth=9
[X] Prop03Sound
     states=10 trans=12 depth=9
[X] Prop04MsgSound
     states=10 trans=12 depth=9
---------- Network05Outbox ----------
[X] Prop01Type
     states=10 trans=12 depth=9
[X] Prop02Safe
     states=10 trans=12 depth=9
[X] Prop03Sound
     states=10 trans=12 depth=9
[X] Prop04MsgSound
     states=10 trans=12 depth=9
---------- Network06Fifo ----------
[X] Prop01Type
     states=10 trans=12 depth=9
[X] Prop02Safe
     states=10 trans=12 depth=9
[X] Prop03Sound
     states=10 trans=12 depth=9
[X] Prop04MsgSound
     states=10 trans=12 depth=9
---------- Network07RSC ----------
[X] Prop01Type
     states=10 trans=12 depth=9
[X] Prop02Safe
     states=10 trans=12 depth=9
[X] Prop03Sound
     states=10 trans=12 depth=9
[X] Prop04MsgSound
     states=10 trans=12 depth=9
done.
